test/functional/outputs/forward_to_elasticsearch_index_test.go

```diff
@@ -0,0 +1,190 @@
+package outputs
+
+import (
+	"fmt"
+
+	"github.com/ViaQ/logerr/log"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	"github.com/openshift/cluster-logging-operator/test/matchers"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+)
+
+var _ = Describe("[Functional][Outputs][ElasticSearch][Index] FluentdForward Output to specific ElasticSearch index", func() {
+
+	const (
+		elasticSearchTag   = "7.10.1"
+		elasticSearchImage = "elasticsearch:" + elasticSearchTag
+		IndexKey           = "indexkey"
+		IndexValue         = "myindex"
+		AppIndex           = "app-write"
+	)
+
+	var (
+		framework                 *functional.FluentdFunctionalFramework
+		addElasticSearchContainer runtime.PodBuilderVisitor
+
+		newVisitor = func(f *functional.FluentdFunctionalFramework) runtime.PodBuilderVisitor {
+			return func(b *runtime.PodBuilder) error {
+				log.V(2).Info("Adding ElasticSearch output container", "name", logging.OutputTypeElasticsearch)
+				b.AddLabels(map[string]string{
+					IndexKey: IndexValue,
+				}).
+					AddContainer(logging.OutputTypeElasticsearch, elasticSearchImage).
+					AddEnvVar("discovery.type", "single-node").
+					AddRunAsUser(2000).
+					End()
+				return nil
+			}
+		}
+
+		// Template expected as output Log
+		outputLogTemplate = functional.NewApplicationLogTemplate()
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+		addElasticSearchContainer = newVisitor(framework)
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	Context("when sending to ElasticSearch "+elasticSearchTag+" protocol", func() {
+		It("should send logs to indexName", func() {
+			IndexName := "myindex"
+			functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+				FromInput(logging.InputNameApplication).
+				ToOutputWithVisitor(func(spec *logging.OutputSpec) {
+					spec.Elasticsearch = &logging.Elasticsearch{
+						IndexName: IndexName,
+					}
+				}, logging.OutputTypeElasticsearch)
+			Expect(framework.DeployWithVisitor(addElasticSearchContainer)).To(BeNil())
+			Expect(framework.WritesApplicationLogs(1)).To(BeNil())
+			raw, err := framework.GetLogsFromElasticSearchIndex(logging.OutputTypeElasticsearch, IndexName)
+			Expect(err).To(BeNil(), "Expected no errors reading the logs")
+			Expect(raw).To(Not(BeEmpty()))
+
+			// Parse log line
+			var logs []types.ApplicationLog
+			err = types.StrictlyParseLogs(raw, &logs)
+			Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+			// Compare to expected template
+			outputTestLog := logs[0]
+			outputLogTemplate.ViaqIndexName = ""
+			Expect(outputTestLog).To(matchers.FitLogFormatTemplate(outputLogTemplate))
+		})
+	})
+	Context("when sending to ElasticSearch "+elasticSearchTag+" protocol", func() {
```

Comment:
It looks like you can remove this context since and pull the "It" into the preceeding one since they are the same